lib/services/chunk_to_items_mapper.py

```diff
@@ -0,0 +1,166 @@
+"""
+Service to map document chunks to Docling items/regions
+"""
+
+import logging
+from typing import Dict, List, Optional, Tuple
+
+from lib.agents.models import ValidatedDocument
+from lib.services.docling_models import (
+    BBox,
+    ChunkToItems,
+    DoclingDocument,
+    DoclingRegion,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _collect_docling_items(doc_dict: Dict) -> List[Tuple[dict, str]]:
+    """Collect all items from Docling document"""
+    items = []
+    for item in doc_dict.get("texts", []):
+        items.append((item, "text"))
+    for item in doc_dict.get("tables", []):
+        items.append((item, "table"))
+    for item in doc_dict.get("pictures", []):
+        items.append((item, "picture"))
+    for item in doc_dict.get("key_value_items", []):
+        items.append((item, "kv"))
+    return items
+
+
+def _extract_item_text(item_dict: dict) -> str:
+    """Extract text content from a Docling item"""
+    if "text" in item_dict:
+        return item_dict["text"]
+    if "content" in item_dict:
+        return item_dict["content"]
+    return ""
+
+
+def _extract_bbox(item_dict: dict) -> Optional[dict]:
+    """Extract bbox data from item, checking multiple locations"""
+    bbox_data = item_dict.get("bbox")
+    if not bbox_data and "prov" in item_dict:
+        prov_list = item_dict.get("prov", [])
+        if prov_list:
+            bbox_data = prov_list[0].get("bbox")
+
+    # Handle bbox as list or dict
+    if isinstance(bbox_data, list) and bbox_data:
+        return bbox_data[0]
+    return bbox_data
```

Comment:
Not sure if it's possible in this scenario, but if you're just converting a dict/json to a pydantic model in these extract functions, maybe the extract functions are not needed at all. You can call a `doc = DoclingDocument.model_validate_json(json_data)` to convert a json that matches the pydantic schema to an instance